src/Microsoft.ML.FastTree/FastTreeCatalog.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.FastTree;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.StaticPipe.Runtime;
+using System;
+
+namespace Microsoft.ML
+{
+    /// <summary>
+    /// FastTree <see cref="TrainContextBase"/> extension methods.
+    /// </summary>
+    public static partial class RegressionTrainers
```

Comment:
There already is a `Microsoft.ML.RegressionTrainers` class in the StandardLearners and LightGBM assemblies. We shouldn't have public types with the same namespace and name across multiple assemblies. It is only going to cause headaches for our users.

Reply:
This applies to all the other extension classes as well.

---
In reply to: [226109180](https://github.com/dotnet/machinelearning/pull/1252#discussion_r226109180) [](ancestors = 226109180)

Reply:
I believe this was intentionally done by Senja for some documentation reasons?

---
In reply to: [226109272](https://github.com/dotnet/machinelearning/pull/1252#discussion_r226109272) [](ancestors = 226109272,226109180)

Reply:
Let me keep it as is for now, and I'll sync with her and understand why it was done. For me, it would've been much easier to group by trainer, into `FastTreeTrainerCatalog` or something

---
In reply to: [226109747](https://github.com/dotnet/machinelearning/pull/1252#discussion_r226109747) [](ancestors = 226109747,226109272,226109180)

Reply:
> Let me keep it as is for now

I don't know .... this is a pretty strong "don't" rule in general. If we are going to fix this very soon (like the next few days), I'm OK with it. But please log an issue. We can't ship with conflicting types across our assemblies.

Reply:
OK spoke to Senja, we should revert that :)

---
In reply to: [226113726](https://github.com/dotnet/machinelearning/pull/1252#discussion_r226113726) [](ancestors = 226113726)